patches/minecraft/net/minecraft/block/BlockCocoa.java.patch

```diff
@@ -1,6 +1,31 @@
 --- ../src-base/minecraft/net/minecraft/block/BlockCocoa.java
 +++ ../src-work/minecraft/net/minecraft/block/BlockCocoa.java
-@@ -131,7 +131,14 @@
+@@ -43,13 +43,19 @@
+         {
+             this.func_176500_f(p_180650_1_, p_180650_2_, p_180650_3_);
+         }
+-        else if (p_180650_1_.field_73012_v.nextInt(5) == 0)
++        else
+         {
+-            int i = ((Integer)p_180650_3_.func_177229_b(field_176501_a)).intValue();
+-
+-            if (i < 2)
++            net.minecraftforge.event.world.BlockEvent ev = new net.minecraftforge.event.world.BlockEvent.CropGrowEvent.Pre(p_180650_1_,p_180650_2_,p_180650_3_);
++            net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(ev);
++            if(ev.getResult() == net.minecraftforge.fml.common.eventhandler.Event.Result.ALLOW || (ev.getResult() == net.minecraftforge.fml.common.eventhandler.Event.Result.DEFAULT && p_180650_1_.field_73012_v.nextInt(5) == 0))
+             {
+-                p_180650_1_.func_180501_a(p_180650_2_, p_180650_3_.func_177226_a(field_176501_a, Integer.valueOf(i + 1)), 2);
++                int i = ((Integer)p_180650_3_.func_177229_b(field_176501_a)).intValue();
++
++                if (i < 2)
++                {
```

Comment:
You tab this section of code in another layer which makes the patches bigger and makes it hard to tell what you add and what is vanilla, fix that.